### ISSUE TITLE
Populate params

### ISSUE THREAD

**Issue by jaysalvat** [loris/api-query-params#98]
**[EDIT] 
I had actually some time and make 
the PR #99 with a POC.**

Hello Loris!
Thanks for sharing this great lib.

It would be great to have a `populate` params to handle relation populations.
[https://mongoosejs.com/docs/populate.html](https://mongoosejs.com/docs/populate.html)

Ex. Passing `?populate=users` to the query, would get:

```javascript
  const { filter, skip, limit, sort, projection, population } = aqp(req.query);
  User.find(filter)
    .skip(skip)
    .limit(limit)
    .sort(sort)
    .select(projection)
    .populate(population) // <---- here
    .exec((err, users) => {
      if (err) {
        return next(err);
      }
 
      res.send(users);
    });
```

... would turn that

```javascript
{
  "_id": "5ce158dd5e60d70972ff97dc",
  "name": "admins",
  "users": [
    "5cded210825b89040cb81185",
    "5cdebeb003db96001fabd040",
  ]
}
````

in that:

```javascript
{
  "_id": "5ce158dd5e60d70972ff97dc",
  "name": "admins",
  "users": [
    { 
      "_id": "5cded210825b89040cb81185",
      "firstname": "brad",
      "lastname": "pitt"
    },
    {
      "_id": "5cdebeb003db96001fabd040",
      "firstname": "matt",
      "lastname": "damon"
    }
  ]
}
```

Since this lib is already handling projection, it could handle population projection:
Ex. `?populate=users&field=name,users.lastname`

# TL;DR;

```
?populate=a,b,c

{ 
  population: [ 'a', 'b', 'c' ];
}
```

```
?populate=a,b,c?fields=a.f1,a.f2,b.f3.c.f4

{
  population: [
    { path: 'a', select: [ 'f1', 'f2' ]},
    { path: 'b', select: [ 'f3' ]},
    { path: 'c', select: [ 'f4' ]}
  ]
}
```

What do you think ?

**Comment by loris** [comment]
Hello Jay,

Thanks a lot for the issue & PR, this looks great! It's already merged, I will update the README and publish a new version later today.
`api-query-params` does not support populate until today because it is a mongoose only feature and I tried to keep it agnostic. But since most node.js+mongo devs seems to use mongoose anyway, let's do it :)

**Comment by jaysalvat** [comment]
Hello Loris,
I though exactly the same (agnostic)... Then I saw your code example with Moongose and I said to myself "Too bad to not have this feature since most node/mongo devs use Mongoose. Let's do it" :)
After all, nothing is broken and keep working as expected. Just a new feature added.

An idea would be to pass a flag `aqp(req.query, MONGOOSE_READY);` to return Mongoose only features. But is it really necessary... :)

Thanks.
